undertow/WFLY-17766_AccessLog_Termination.adoc

```diff
@@ -0,0 +1,114 @@
+= [Community] Add ability to configure access log termination schedule and delay
+:author:            Bartosz Baranowski
+:email:             bbaranow@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This RFE(https://issues.redhat.com/browse/WFLY-17766) is part of bug fix for https://issues.redhat.com/browse/WFLY-13933 / https://issues.redhat.com/browse/JBEAP-20056 . In order to avoid bad user experience and possible follow up, it would be
+good to allow configuration of write termination delay. Fix, in both issues is split into functional commit and second RFE commit - which adds functionality and model changes.
+
+Fix commit include only changes to allow graceful takeover of access log write operation. During runtime write ops are being handled by NIO thread. This thread on shutdown is terminated abruptly due to: https://issues.redhat.com/browse/WFCORE-1632
+To mitigate this problem first commit fix contractual requirements for access log classes and shifts burden of termination from NIO worker - just like other WFLY services that do async close.
+Async close will wait till aloted time( with interim checks ) runs off to take over agressively write OPs or until NIO worker yield. As is, in fix number of interim checks and wait period is fixed(interim_period*number_of_retries).
+
+RFE commits add changes to model and services in order to allow configuration of:
+ - number of retries between NIO worker state polls
```

Comment:
do we really want to configure the number of retries, or just the maximum amount of delay here will be enough to server users? What I would like to know if there is a specific case where the final user might want to tweak the number of retries. Maybe we should come up with an algorithm that infers the number of retries based on the amount of delay that was configured.

Reply:
Granularity?  GIves users better control.